src/main/kotlin/com/bridgecrew/services/CheckovResultsComparatorGenerator.kt

```diff
@@ -15,55 +16,79 @@ class CheckovResultsComparatorGenerator {
         private val LOG = logger<CheckovResultProperty>()
         fun generateCheckovResultComparator(): Comparator<BaseCheckovResult> {
             return generateNameComparator(CheckovResultProperty.FILE_PATH)
-                    .thenComparing(generateNameComparator(CheckovResultProperty.RESOURCE))
+                    .thenComparing(generateResourceComparator())
                     .thenBy { it.severity }
                     .thenComparing(generateNameComparator(CheckovResultProperty.NAME))
         }
 
-        fun generateNameComparator(property: CheckovResultProperty): Comparator<BaseCheckovResult> {
+        fun generateResourceComparator(): Comparator<BaseCheckovResult> {
             return Comparator { result1, result2 ->
-                val name1 = extractNameByProperty(property, result1)
-                val name2 = extractNameByProperty(property, result2)
-                try {
-                    if (name1.equals(name2, true))
-                        return@Comparator 0
+                if (result1.category == Category.SECRETS && result2.category != Category.SECRETS) {
```

Comment:
Maybe a better idea is to add a numeric value to the enum (also applicable to severity for example) and this way we can sort by a numeric value which makes more sense. WDYT?

Reply:
I don't know if this is what you meant but a great idea!